### PR TITLE
Fix test execution harness to actually run the tests, fix data-system-tests, add test-query

### PR DESCRIPTION
**Pull request by philvarner**
- Fix test execution harness to actually run the tests, and report failure if none do
- fix data-system-tests to pass (tests weren't actually failing, just the code no longer ran correctly)
- add Query tests

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,9 @@
 FROM mas.ops.maap-project.org/root/jupyter-image/vanilla:develop
 
+COPY requirements.txt requirements.txt
```
Comment (philvarner) [inline Dockerfile:3]: ipykernel and papermill are already installed in the image, and use the same versions as specifiied in requirements.txt 

pytest is the only library that's not already installed in the image

File: README.md [hunk 2]
```diff
@@ -12,9 +12,13 @@
...
-The GitHub App is used as the authentication principal in GitHub that has been granted
-fine-grained permissions to perform actions in this repo. In this case, the only permission
-granted is to invoke the `workflow_dispatch` Action.
+A GitHub App is used as an authentication principal within GitHub that can be granted
+fine-grained permissions to perform actions within a repo. This allows clients to perform authorized actions in GitHub without needing to use credentials for a specific user.  
+
+The GitHub App [maap-system-tests](https://github.com/settings/apps/maap-system-tests) has been created
+to be used by GitHub Actions in the service repos to invoke the system tests in this repo. 
+In this case, a single permission granted is granted to allow invocation of the `workflow_dispatch`
+Action.
```
Comment (philvarner) [inline README.md:21]: @abarciauskas-bgse added this language per your request

File: README.md [hunk 3]
```diff
@@ -59,7 +63,7 @@ Additionally, these parameters can be passed, through the default values in the
 
 The workflow can be manually invoked when testing it using the following steps.
 
-0. The App is [https://github.com/settings/apps/maap-system-tests](https://github.com/settings/apps/maap-system-tests). It's currently part of Phil's account, but will be moved to the MAAP-Project org as soon as we can coordinate it.
+0. The App is [maap-system-tests](https://github.com/settings/apps/maap-system-tests) and is part of the MAAP-Project org.
```
Comment (philvarner) [inline README.md:66]: @abarciauskas-bgse updated this

File: tests/harness.py [hunk 4]
```diff
@@ -44,8 +44,12 @@ def main():
         for (k, v) in failure_results.items():
             print(f"File {k} => {v}")
         exit(1)
+    elif not success_results:
```
Comment (philvarner) [inline tests/harness.py:47]: I should have known better here -- fail if no tests run

File: README.md (hunk 3 above)
Comment (abarciauskas-bgse) [inline README.md:66]:
```suggestion
0. [maap-system-tests](https://github.com/settings/apps/maap-system-tests) is a "github app" of the MAAP-Project org.
```

File: tests/data-system-tests.ipynb [hunk 5]
```diff
@@ -159,8 +151,8 @@
    "outputs": [],
    "source": [
     "# Test endpoints\n",
-    "assert maap._SEARCH_GRANULE_URL == f'https://api.{env_prefix}maap-project.org/api/cmr/granules'\n",
-    "assert maap._SEARCH_COLLECTION_URL == f'https://api.{env_prefix}maap-project.org/api/cmr/collections'\n",
+    "assert maap._SEARCH_GRANULE_URL == f'https://{maap_host}/api/cmr/granules'\n",
```
Comment (abarciauskas-bgse) [inline tests/data-system-tests.ipynb:154]: Minor but maybe rename this notebook to "test-cmr.ipynb" or "test-search.ipynb" to match the other notebook and make it more clear what is being tested in this file

**Review by abarciauskas-bgse — APPROVED** [review]

**Review by tracetechnical — APPROVED** [review]

**Comment by philvarner** [comment]
lol, this test fails now because Mark deleted the dataset yesterday 🙃

**Pull request merged**